lib/api/apiUtils/bucket/getNotificationConfiguration.js

```diff
@@ -0,0 +1,33 @@
+const { errors, models } = require('arsenal');
+const { NotificationConfiguration } = models;
+
+const { config } = require('../../../Config');
+
+function getNotificationConfiguration(parsedXml) {
+    const targets = [];
+    if (!config.bucketNotificationDestinations) {
+        return { error: errors.InvalidArgument.customizeDescription(
+            'Unable to validate the following destination configurations') };
+    }
+    config.bucketNotificationDestinations.forEach(t => {
+        targets.push(t.resource);
+    });
```

Comment:
Another for each and push.
```suggestion
    if (!config.bucketNotificationDestinations) {
        return { error: errors.InvalidArgument.customizeDescription(
            'Unable to validate the following destination configurations') };
    }
    const targets = config.bucketNotificationDestinations.map(t => t.resource);
```

Reply:
Alternatively, targets can be a set to make the lookup below O(1) instead of O(n).

```suggestion
    if (!config.bucketNotificationDestinations) {
        return { error: errors.InvalidArgument.customizeDescription(
            'Unable to validate the following destination configurations') };
    }
    const targets = new Set(config.bucketNotificationDestinations.map(t => t.resource));
```